conan/tools/microsoft/toolchain.py

```diff
@@ -27,6 +28,21 @@ def _name_condition(settings):
         return name.lower(), condition
 
     def write_toolchain_files(self):
+        # Warning
+        msg = ("\n*****************************************************************\n"
+               "******************************************************************\n"
+               "'write_toolchain_files()' has been deprecated and moved.\n"
+               "It will be removed in next Conan release.\n"
+               "Use 'generate()' method instead.\n"
+               "********************************************************************\n"
+               "********************************************************************\n")
+        from conans.client.output import Color, ConanOutput
+        ConanOutput(self._conanfile.output._stream,
+                    color=self._conanfile.output._color).writeln(msg, front=Color.BRIGHT_RED)
```

Comment:
Why not...?  (It is ok because we will remove all of this in the next release)

```python
self._conanfile.output.warn(msg)
```

Reply:
Because it only prints in RED color the package name and version, so the message is not that visible, as it is printed in regular white, together with a wall of text. I wanted to make it super-visible and clear.

This is something that we might want to change for warnings and errors in our output, I'd say the UX is not the best (easy to miss important warnings or messages)

Reply:
It prints everything in yellow/red, it is the same call (with a prefix and output to stderr):

```python
    def warn(self, data):
        self.writeln("WARN: {}".format(data), Color.BRIGHT_YELLOW, error=True)

    def error(self, data):
        self.writeln("ERROR: {}".format(data), Color.BRIGHT_RED, error=True)
```

Reply:
``conanfile.output`` is of type ``ScopedOutput`` and it will only print the scope in color, not the whole text.